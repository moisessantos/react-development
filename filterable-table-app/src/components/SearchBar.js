import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SearchBar extends Component {
  constructor(){
    super();
    this.filter = this.filter.bind(this);
  }

  render() {
    return (
      <form className="container">
        <input type="text" placeholder="Search..." onChange={e => this.filter(e)}/>
        <p>
          <input type="checkbox" onClick={this.props.onClick} />
          {' '}
          Only show products in stock
        </p>
      </form>
    );
  }
  
  filter(e) {
    this.props.filter(e.target.value);
  }
}

SearchBar.propTypes = {
  filter: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
};

export default SearchBar;
