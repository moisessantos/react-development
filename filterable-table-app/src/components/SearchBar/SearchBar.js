import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SearchBar extends Component {
  constructor(){
    super();
    this.filter = this.filter.bind(this);
  }

  render() {
    return (
      <form className="container">
        <input type="text" placeholder="Search..." onChange={e => this.filter(e)}/>
        <p>
          <input type="checkbox" onClick={this.props.handleClick} />
          {' '}
          Only show products in stock
        </p>
      </form>
    );
  }
  
  filter(e) {
    this.props.handleFilter(e.target.value);
  }
}

SearchBar.propTypes = {
  handleFilter: PropTypes.func.isRequired,
  handleClick: PropTypes.func.isRequired
};

export default SearchBar;
