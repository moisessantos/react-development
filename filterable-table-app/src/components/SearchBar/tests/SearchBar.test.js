import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../';

describe('SearchBar component', () => {
  const handleClick = jest.fn(), handleFilter= jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  const shallowComponent = () => shallow(<SearchBar handleClick = { handleClick } handleFilter = { handleFilter } />);

  it('should render snapshot', () => {
    const wrapper = shallowComponent();
    expect(wrapper).toMatchSnapshot();
  });
  it('should trigger handleFilter when input changes', () => {
    const expectedValue = 'ball';
    const wrapper = shallowComponent();
    
    wrapper.find('input[type="text"]').simulate('change', { target: { value: expectedValue } });

    expect(handleFilter).toHaveBeenCalledTimes(1);
    expect(handleFilter).toHaveBeenCalledWith(expectedValue);
  });
  it('should trigger onChange when input is checked', () => {
    const wrapper = shallowComponent();
    
    wrapper.find('input[type="checkbox"]').simulate('click');

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
