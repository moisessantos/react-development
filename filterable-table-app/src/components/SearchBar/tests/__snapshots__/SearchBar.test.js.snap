// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should render snapshot 1`] = `
<form
  className="container"
>
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
  />
  <p>
    <input
      onClick={[MockFunction]}
      type="checkbox"
    />
     
    Only show products in stock
  </p>
</form>
`;
